src/screens/AddressDetail.js

```diff
@@ -0,0 +1,359 @@
+
+        const newNumberOfTransactions = this.state.numberOfTransactions + 1;
+
+        // Finally we update the state again
+        this.setState({ transactions, hasAfter, numberOfTransactions: newNumberOfTransactions });
+      }
+    }
+  }
+
+  /**
+   * Check if address is valid and then update the state and get data from full node
+   * If not valid show error message
+   *
+   * @param {Object} address New searched address to update state
+   */
+  updateAddress = (address) => {
+    if (transaction.isAddressValid(address)) {
+      this.setState({ address, errorMessage: '' }, () => {
+        const queryParams = this.pagination.obtainQueryParams();
+        if (queryParams.token !== null) {
+          // User already have a token selected on the URL
+          this.setState({ selectedToken: queryParams.token }, () => {
+            this.getSummaryData();
+            this.getHistoryData(this.state.queryParams);
+          });
+        } else {
+          // Will get data and select the default token
+          // In this case I don't get history data because I still don't know the token
+          // When the token changes, I will fetch the history data
+          this.getSummaryData();
+        }
+      });
+    } else {
+      this.setState({ errorMessage: 'Invalid address.' });
+    }
+  }
+
+  /**
+   * Update transactions data state after requesting data from the server
+   *
+   * @param {Object} queryParams URL parameters
+   */
+  getHistoryData = (queryParams) => {
+    hathorLib.walletApi.getSearchAddress(this.state.address, TX_COUNT, queryParams.hash, queryParams.page, queryParams.token, (response) => {
+      if (response.success) {
+        this.handleFetchedData(response, queryParams);
+      }
+    });
+  }
+
+
+  /**
+   * Update component state when new list data arrives
+   *
+   * @param {Object} data Response data from the server
+   * @param {Object} queryParams URL parameters
+   */
+  handleFetchedData = (data, queryParams) => {
+    // Handle differently if is the first GET response we receive
+    // page indicates if was clicked 'previous' or 'next'
+    // Set first and last hash of the transactions
+    let hasAfter;
+    let hasBefore;
+    if (queryParams.page === 'previous') {
+      hasAfter = true;
+      hasBefore = data.has_more;
+      if (!hasBefore) {
+        // Went back to most recent page: clear URL params
+        this.pagination.clearOptionalQueryParams();
+      }
+    } else if (queryParams.page === 'next') {
+      hasBefore = true;
+      hasAfter = data.has_more;
+    } else {
+      // First load without parameters
+      hasBefore = false;
+      hasAfter = data.has_more;
+    }
+
+    this.setState({
+      transactions: data.transactions,
+      loadingHistory: false,
+      hasAfter,
+      hasBefore,
+      queryParams,
+      numberOfTransactions: data.total,
+    });
+  }
+
+  /**
+   * Request data from server and update state balance
+   */
+  getSummaryData() {
```

Comment:
All other methods use es6 style functions. Weird that this works without `bind` 🤔

Reply:
Yes, it's weird. It was working fine. But changed it anyway.